src/screens/runPlan/runPlanList/PlanItemListItem.tsx

```diff
@@ -1,47 +1,67 @@
 import React from 'react';
 import {StyleSheet, TouchableHighlight, ViewStyle} from 'react-native';
 
+import { NavigationService } from '../../../services';
 import { Card } from 'components';
-import { PlanItem } from 'models';
-import {palette, typography} from 'styles';
+import { Plan, PlanItem, PlanSubItem, Student } from 'models';
+import { palette } from 'styles';
 import { PlanItemName } from '../PlanItemName';
+import { Function } from '@babel/types';
 
 interface Props {
-  planItem: PlanItem;
+  student: Student;
+  itemParent: PlanItem | Plan;
+  item: PlanItem | PlanSubItem;
   index: number;
-  textSize: string;
-  textCase: string;
   currentTaskIndex: number;
 }
 
 export class PlanItemListItem extends React.PureComponent<Props> {
   textContainer(): ViewStyle {
-    return this.props.planItem.completed ? styles.textContainerCompleted : styles.textContainer;
+    return this.props.item.completed ? styles.textContainerCompleted : styles.textContainer;
   }
 
-  nameTextColor() {
-    return this.props.planItem.completed ? styles.nameTextColorCompleted : styles.nameTextColor;
+  nameTextColor(): ViewStyle {
+    return this.props.item.completed ? styles.nameTextColorCompleted : styles.nameTextColor;
   }
 
   markItemPlanAsCompleted = () => {
     if (this.props.index === this.props.currentTaskIndex) {
-    this.props.planItem.update({
-      completed: true,
+      if(this.props.item instanceof PlanItem) {
```

Comment:
Going to next files I see that we need for sure to remove this condition which binds our logic to a particular Item type and put some abstraction here. But as I wrote at the beginning I think we should leave it for now, and I will try to update this during next week refactoring (cause it's tightly related with current models composition).